distributed/client.py

```diff
@@ -5273,6 +5287,11 @@ def __init__(self, futures=None, loop=None, with_results=False, raise_errors=Tru
         self.with_results = with_results
         self.raise_errors = raise_errors
 
+        if timeout is not None:
```

Comment:
Would it make sense to re-use `distributed.utils.Deadline` here?

Reply:
yep, done